spec/jobs/reports/verification_errors_report_spec.rb

```diff
@@ -0,0 +1,116 @@
+require 'rails_helper'
+
+describe Reports::VerificationErrorsReport do
+  let(:issuer) { 'urn:gov:gsa:openidconnect:sp:sinatra' }
+  let(:email) { 'foo@bar.com' }
+  let(:name) { 'An SP' }
+  let(:user) { create(:user) }
+  let(:uuid) { 'foo' }
+  let(:user2) { create(:user) }
+  let(:uuid2) { 'foo2' }
+  let(:header) { "uuid,welcome_view_at,error_code\r\n" }
+
+  subject { described_class.new }
+
+  it 'is does not send out an email with nothing configured' do
+    expect(UserMailer).to_not receive(:verification_errors_report)
+
+    subject.perform(Time.zone.today)
+  end
+
+  it 'sends out a blank report if no users went through doc auth' do
+    run_report_and_expect(header)
+  end
+
+  it 'sends out an abandon code on a user lands on welcome and leaves' do
+    now = Time.zone.now
+    DocAuthLog.create(user_id: user.id, welcome_view_at: now, issuer: issuer)
+
+    run_report_and_expect("#{header}#{uuid},#{now.utc},ABANDON\r\n")
+  end
+
+  it 'sends out a blank report if no issuer data' do
+    now = Time.zone.now
+    DocAuthLog.create(user_id: user.id, welcome_view_at: now, issuer: 'issuer2')
+
+    run_report_and_expect(header)
+  end
+
+  it 'sends out a document error if the user submits document but does not progress forward' do
+    now = Time.zone.now
+    DocAuthLog.create(
+      user_id: user.id,
+      welcome_view_at: now,
+      document_capture_submit_at: now + 1.second,
+      issuer: issuer,
+    )
+
+    run_report_and_expect("#{header}#{uuid},#{now.utc},DOCUMENT_FAIL\r\n")
```

Comment:
I think we could make the tests clearer if we parsed the CSV result...

```suggestion
    csv = CSV.parse(result)
    expect(csv.first['uuid']).to eq(uuid)
    expect(csv.first['timestamp']).to eq(now.utc)
    expect(csv.first['status']).to eq('DOCUMENT_FAIL')
```

Reply:
The tests might be clearer but then you are not testing the format of the output because the ruby CSV might be forgiving on malformed CSV

Reply:
If Ruby CSV isn't parsing a CSV correctly, we have other major problems.... I think it's important to trust the standard library

Reply:
for what it's worth, we do `CSV.parse` in some existing tests, so I think it's fine to trust `CSV` to parse properly here as well.

Are instances where we know Ruby parses CSVs incorrectly that would impact this usage?

Reply:
correctly is a relative term.  some libraries will throw an error on invalid values or rows others will skip it which can also be dependent on your settings.

Reply:
I'll change it but to me it's like saying we have a spec to build a 15 gallon gas tank and the way we test it is to see if we can drive 300 miles.

Reply:
Let's do it for consistency with rest of our tests:
```
> git grep -l CSV.parse
app/services/user_seeder.rb
lib/tasks/create_test_accounts.rb
spec/jobs/reports/agreement_summary_report_spec.rb
spec/jobs/reports/combined_invoice_supplement_report_spec.rb
spec/jobs/reports/daily_dropoffs_report_spec.rb
spec/jobs/reports/total_ial2_costs_report_spec.rb
spec/services/data_requests/write_user_events_spec.rb
```

Reply:
ok by that same logic we should keep the translations :-)